### PR TITLE
remove trigger if subscription disabled

### LINKED ISSUE #857: Subscription check notification trigger
https://github.com/subquery/subql/blob/1a719fb26cc4cd04900122e3918a27c062b2afb6/packages/node/src/utils/sync-helper.ts#L42

Notification trigger should never been replaced.

- [x] remove trigger if unsafe disabled

### LINKED ISSUE #872: Remove notification trigger if subscription is disabled

### PR DESCRIPTION
**Pull request by jiqiang90**
Closes #857

### CODE REVIEW

**Comment by jiqiang90** [comment]
https://github.com/subquery/subql/blob/1a719fb26cc4cd04900122e3918a27c062b2afb6/packages/node/src/utils/sync-helper.ts#L42

In https://github.com/subquery/subql/issues/857 we mentioned the function should never been replaced. 

Not sure how to make it `CREATE IF NOT EXISTS FUNCTION send_notification()` here.

Also check the function again, seems it is harmless, any thought? @stwiname @ianhe8x

**Review by stwiname — APPROVED** [review]
Replace function seems fine to me. I'm not an expert at this though

File: packages/node/src/utils/sync-helper.ts [hunk 1]
```diff
@@ -67,10 +67,14 @@ BEGIN
...
+export function dropNotifyTrigger(schema: string, table: string): string {
+  return `DROP TRIGGER IF EXISTS "${schema}_${table}_notify_trigger"
+    ON "${schema}"."${table}";`;
+}
+
 export function createNotifyTrigger(schema: string, table: string): string {
   return `
-DROP TRIGGER IF EXISTS "${schema}_${table}_notify_trigger"
-    ON "${schema}"."${table}";
+${dropNotifyTrigger(schema, table)}
```
Comment (ianhe8x) [inline packages/node/src/utils/sync-helper.ts:77]: Do you plan to improve here? Every restart will drop and recreate trigger doesn't seems a good way.

Comment (jiqiang90) [reply packages/node/src/utils/sync-helper.ts:77]: Seems there is no `CREATE IF NOT EXISTS` on trigger.

**Review by stwiname — APPROVED** [review]

File: packages/node/src/utils/sync-helper.ts [hunk 2]
```diff
@@ -72,9 +72,12 @@ export function dropNotifyTrigger(schema: string, table: string): string {
     ON "${schema}"."${table}";`;
 }
 
+export function getNotifyTriggers(schema: string, table: string): string {
+  return `select trigger_name as "triggerName", event_manipulation as "eventManipulation" from information_schema.triggers
```
Comment (ianhe8x) [inline packages/node/src/utils/sync-helper.ts:76]: Use query parameter to avoid sql injection

**Review by ianhe8x — APPROVED** [review]

**Pull request merged**